language/move-prover/docgen/src/docgen.rs

```diff
@@ -664,7 +691,179 @@ impl<'env> Docgen<'env> {
         }
     }
 
-    /// Gen header for TOC, returning label where we can later insert the content after
+    /// Generate a static call diagram (.svg) starting from the given function.
+    fn gen_call_diagram(&self, fun_id: QualifiedId<FunId>) {
+        let fun_env = self.env.get_function(fun_id);
+        let name_of = |env: &FunctionEnv| {
+            if fun_env.module_env.get_id() == env.module_env.get_id() {
+                env.get_simple_name_string()
+            } else {
+                Rc::from(format!("\"{}\"", env.get_name_string()))
+            }
+        };
+
+        let mut dot_src: String = String::from("digraph G {\n");
+        let mut visited: BTreeSet<QualifiedId<FunId>> = BTreeSet::new();
+        let mut queue: VecDeque<QualifiedId<FunId>> = VecDeque::new();
+
+        visited.insert(fun_id);
+        queue.push_back(fun_id);
+
+        while let Some(id) = queue.pop_front() {
+            let caller_env = self.env.get_function(id);
+            let caller_name = name_of(&caller_env);
+            let callee_list = caller_env.get_called_functions();
+
+            if fun_env.module_env.get_id() == caller_env.module_env.get_id() {
+                dot_src.push_str(&format!("    {}\n", caller_name));
```

Comment:
Might be personal preference, but what would you think of using `\t` instead of the fully written-out spaces in these strings?

Reply:
Done.